Fabric/src/main/java/me/ferdz/placeableitems/block/PlaceableItemsBlock.java

```diff
@@ -191,6 +184,19 @@ public Item asItem() {
         return this.item;
     }
 
+    // Used for block placement in the ItemPlaceHandler
+    public BlockItem getBlockItem() {
+        return getDefaultBlockItem();
+    }
+
+    protected BlockItem getDefaultBlockItem() {
+        if (blockItem == null) {
+            this.blockItem = new BlockItem(this, new Item.Settings());
+        }
+
+        return blockItem;
+    }
```

Comment:
I would rather see all of this in a single method. This kind of implies that `getBlockItem` should be overriden, but the point of the component system is to not require extensions for block implementations

Reply:
That's something that didn't come to mind, you're right. In truth, I'm not sure there's any rhyme or reason to be overriding this functionality anyways.